gpu/gpu.go

```diff
@@ -248,6 +300,9 @@ func CheckVRAM() (int64, error) {
 			overhead = gpus * 1024 * 1024 * 1024
 		}
 		avail := int64(gpuInfo.FreeMemory - overhead)
+		if (gpuInfo.Library == "cuda" && CudaTegra != "") {
+			avail = int64(gpuInfo.FreeMemory)
```

Comment:
Tegra has a caching system it uses that eats up "available memory" but is ready to be freed on demand. When I left in the 1024MB overhead, it would frequently give "falling back to CPU -> OOM" even though JTOP was showing enough memory free. I tried to find an automated system to properly report available memory to include freeable cache and the gist is that "free -m" produces the most accurate (and accurate is almost an incorrect word) depiction of free memory in Linux and should be what is used. There isn't a library or API to query the same data other than reading it directly from /proc (and in fact, that's what the `free` binary does). 

TL;DR I let the Tegra OS manage overhead as it seems to be better at it than me. Also the reported memory from the sysinfo library, the cuda driver free memory reported, and jtop were all inconsistent in the values reported.